### PR TITLE
[CSL-1763] Add support for type in tasks

### PR DESCRIPTION
**Pull request by stanlp1**
Adjusted test for start and end date to use last 30 days instead of hard coded date.

### CODE REVIEW

File: constructorio-client/src/main/java/io/constructor/client/TaskRequest.java [hunk 1]
```diff
@@ -1,8 +1,5 @@
 package io.constructor.client;
 
-import java.io.File;
```
Comment (jjl014) [inline constructorio-client/src/main/java/io/constructor/client/TaskRequest.java:3]: Nice! 😄

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIOTasksTest.java [hunk 2]
```diff
@@ -121,8 +122,10 @@ public void AllTasksShouldReturnAListOfTasksWhenStartAndEndDateIsPassed() throws
         ConstructorIO constructor = new ConstructorIO(apiToken, apiKey, true, null);
         AllTasksRequest request = new AllTasksRequest();
 
-        request.setStartDate("2022-09-03");
-        request.setEndDate("2022-09-30");
+        LocalDate endDate = LocalDate.now();
```
Comment (jjl014) [inline constructorio-client/src/test/java/io/constructor/client/ConstructorIOTasksTest.java:125]: Love this change!

**Review by jjl014 — CHANGES_REQUESTED** [review]
Nice! The code looks good to me, but it looks like there are 2 failing tests. Might be unrelated to your changes, but we should fix them 👍 

```
newShouldReturnTaskRequestWithDefaults(io.constructor.client.AllTasksRequestTest): expected:<20> but was:<0>
settersShouldSet(io.constructor.client.AllTasksRequestTest): expected:<20> but was:<0>
```

**Review by jjl014 — COMMENTED** [review]
Thanks for fixing the tests. Not sure what happened or something in the endpoint changed, but it looks like more tests are failing now (different tests). o_O

```
TaskRequestWithInvalidApiKeyShouldError(io.constructor.client.ConstructorIOTaskTest): (..)

TaskAsJSONRequestWithInvalidApiKeyShouldError(io.constructor.client.ConstructorIOTaskTest): (..)

AllTasksShouldReturnErrorWithInvalidApiKey(io.constructor.client.ConstructorIOTasksTest): (..)

AllTasksAsJSONShouldReturnErrorWithInvalidApiKey(io.constructor.client.ConstructorIOTasksTest): (..)
  ```

**Comment by stanlp1** [comment]
Weirdly things still pass for me. Could it be you got rate limited for those times?

**Review by jjl014 — APPROVED** [review]
Lgtm! 🚀

**Pull request merged**